darwin/future/core/client.py

```diff
@@ -2,21 +2,27 @@
 
 from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import Callable, Dict, List, Optional, Union, overload
+from typing import Any, Callable, Dict, List, Optional, Union, overload
 from urllib.parse import urlparse
 
 import requests
-from pydantic import BaseModel, validator
+import yaml
+from pydantic import BaseModel, root_validator, validator
 from requests.adapters import HTTPAdapter, Retry
 
 from darwin.future.core.types.query import Query
 from darwin.future.data_objects.darwin_meta import Team
 from darwin.future.exceptions.client import NotFound, Unauthorized
 
-HTTPMethod = Union[Callable[[str], requests.Response], Callable[[str, dict], requests.Response]]
+JSONType = Dict[str, Any]  # type: ignore
 
 
-class Config(BaseModel):
+class TeamsConfig(BaseModel):
+    api_key: str
+    datasets_dir: Path
+
+
+class DarwinConfig(BaseModel):
```

Comment:
Had to change from Config -> DarwinConfig here because pydantic models actually have an internal Config class which sets metaproperties and it causes mypy issues on functions like local() which return the 'Config' class. It can't figure out if you mean the base class or the internal class.